timeless/reservations/views.py

```diff
@@ -116,37 +117,43 @@ def list():
     return render_template("restaurants/tables/list.html")
 
 
-@BP.route("/create", methods=("GET", "POST"))
-def create():
-    """ Create new reservation """
-    """
-    @todo #380:30min Continue. When we use the form.validate() function the
-     status (enum: timeless/restaurants/models.py) is displaying the
-     following error: 'status': ['Invalid Choice: could not coerce', 'Not a
-     valid choice']. Removes the commented if to see the error. Also refactor
-     the function using genericViews, as explained in puzzles #134 and #137.
-    """
-    form = ReservationForm(request.form)
-    try:
-        """if request.method == "POST" and form.validate():"""
-        if request.method == "POST":
-            reservation = Reservation(
-                start_time=request.form["start_time"],
-                end_time=request.form["end_time"],
-                num_of_persons=request.form["num_of_persons"],
-                comment=request.form["comment"],
-                status=request.form["status"]
-            )
-            DB.session.add(reservation)
-            DB.session.commit()
-            return redirect(url_for("reservations.list"))
-        else:
-            flash("Error: ", form.errors)
-    except Exception as error:
-        flash("Error: ", error)
-
-    return render_template("reservations/create_edit.html", error=form.errors,
-                            action="create", form=form)
+class CreateReservation(views.CrudAPIView):
+    """ Create a new reservation instance """
+
+    def post(self):
+        """ Create new reservation """
+        """
+        @todo #434:30min Continue the implementation of CreateReservation. 
+         Add authentication and refactor using CrudAPIView also edit and 
+         delete methods for Reservations. Tests for these are skipped.
+        """
+        form = ReservationForm(request.form)
+        try:
+            if form.validate():
+                reservation = Reservation(
+                    start_time=request.form["start_time"],
+                    end_time=request.form["end_time"],
+                    num_of_persons=request.form["num_of_persons"],
+                    comment=request.form["comment"],
+                    status=request.form["status"]
+                )
+                DB.session.add(reservation)
+                DB.session.commit()
+                return jsonify(status="success"), HTTPStatus.OK
+
+            return jsonify(
+                status="error",
+                errors=form.errors
+            ), HTTPStatus.BAD_REQUEST
+
+        except Exception as error:
+            return jsonify(
+                status="error",
+                errors=errors
```

Comment:
@emilianodellacasa I think you want to say ``errors=error`` here (singular, since that's the variable name in the ``except``

Reply:
@amihaiemil Correct, thanks